### ISSUE TITLE
com.gargoylesoftware.htmlunit.ElementNotFoundException: elementName=[form] attributeName=[name] attributeValue=[docSearch]

### ISSUE THREAD

**Issue by AbdenourAchab** [HtmlUnit/htmlunit#138]
I am trying to get to the docSearch form of the https://eagletw.mohavecounty.us/treasurer/treasurerweb/search.jsp web page using the latest HTMLUnit release (2.37.0). As you can see using Firefox's DOM Inspector, there is such a form
`WebClient webClient = new WebClient(BrowserVersion.FIREFOX_68);
webClient.getOptions().setThrowExceptionOnScriptError(false);
webClient.setJavaScriptErrorListener(new SilentJavaScriptErrorListener());   
HtmlPage page = webClient.getPage("https://eagletw.mohavecounty.us/treasurer/treasurerweb/search.jsp");
webClient.waitForBackgroundJavaScriptStartingBefore(1000000000);
page =  (HtmlPage) page.getEnclosingWindow().getEnclosedPage();
HtmlForm form = page.getFormByName("docSearch");`

The last line of the above code gives me the following exception:
com.gargoylesoftware.htmlunit.ElementNotFoundException: elementName=[form] attributeName=[name] attributeValue=[docSearch]

This issue was discussed and investigated at 
https://stackoverflow.com/questions/60148335/com-gargoylesoftware-htmlunit-elementnotfoundexception-elementname-form-attri?noredirect=1#comment106625029_60148335

But it was NOT solved in those discussions and investigations.

Here is a summary of what we found out:
1) docSearch does not show up in page.asXml()
2) There are two forms in https://eagletw.mohavecounty.us/treasurer/treasurerweb/search.jsp , and page.getForms() does return two forms. But the second form (the one of interest here) doesn't work. Each of the following pieces of code results in an exception:
2.a) 
`HtmlForm form = page.getForms().get(1);
HtmlInput hi = form.getInputByValue("Search");`
results in the following exception: com.gargoylesoftware.htmlunit.ElementNotFoundException: elementName=[input] attributeName=[value] attributeValue=[Search] 
2.b) `HtmlForm form = page.getForms().get(1);
List<HtmlInput> his = form.getByXPath("//input[@value='Search']"); 
Iterator hisIterator = his.iterator(); HtmlInput hi = (HtmlInput) hisIterator.next();`
results in the following exception:  java.util.NoSuchElementException

**Comment by rbri** [comment]
Ok, spend some time to install a vpn to be able to access the server.
But i need an account to test/reproduce your problem.
Can you please offers some credentials - e.g. via private mail.

**Comment by rbri** [comment]
Have now a VPN up and running, will try to create some sample code.

**Comment by rbri** [comment]
Have added a working sample code to the stackoverflow question. Please try

**Comment by AbdenourAchab** [comment]
The code worked. I added a comment to the stackoverflow question. Copy/pasting part of it here: I am still a little confused though. Your code seems to have logged into something. Why was that necessary, since no logging in is needed when you go to eagletw.mohavecounty.us/treasurer/treasurerweb/search.jsp using Firefox ?

**Comment by rbri** [comment]
The point is the cookie persistence usually done by the browser. HtmlUnit stores cookies only per WebClient instance. Every time you start a new WebClient you start with an empty cookie store. Clear all the cookies in your browser and you have to login with the browser also.

**Comment by AbdenourAchab** [comment]
I just verified what you said. I cleared the cookies on my Firefox for the https://eagletw.mohavecounty.us/treasurer/treasurerweb/search.jsp web page and, indeed, I had to login (even if it's just clicking on the Login button with no username and no password) to get to the forms I was trying to access through WebClient. Now I think I understand what GamingFelix was saying in the StackOverflow discussion.

So, the problem is NOT with HtmlUnit. Closing this issue. Thanks rbri and GamingFelix for your help.
